### DATA 604: Individual Milestone 
#### Danae McCulloch 

The discussions are found further down as I first began my assignment by cleaning the data and creating the necessary SQL tables that are need to write the queries. 

In [260]:
import pandas as pd
import sqlalchemy as sq
import pymysql

In [3]:
username = 'root'
password = 'x'
host = 'localhost'  
port = '3306'       
database = 'Milestone'

# Create the connection URL
connection_url = f"mysql+pymysql://{username}:{password}@{host}:{port}/{database}"

engine = sq.create_engine(connection_url)

In [5]:
#bringing in datasets 

hypertension =  pd.read_csv('Global_Hypertension.csv', encoding='ISO-8859-1')
cholesterol = pd.read_csv('Global_Cholesterol.csv', encoding='ISO-8859-1')
mortality = pd.read_csv('Global_Moratlity.csv', encoding='ISO-8859-1')

# Part 0

### Link to Datasets

The following links access the datasets:

- Hypertension Dataset: https://www.who.int/data/gho/data/indicators/indicator-details/GHO/prevalence-of-hypertension-among-adults-aged-30-79-years
- Cholesterol Dataset:  https://www.who.int/data/gho/data/indicators/indicator-details/GHO/mean-total-cholesterol-(age-standardized-estimate)
- Mortality Dataset: https://www.who.int/data/gho/data/indicators/indicator-details/GHO/number-of-deaths-attributed-to-non-communicable-diseases-by-type-of-disease-and-sex
- World Health Organization. (n.d.). Noncommunicable diseases: Risk factors. WHO. 			   https://www.who.int/data/gho/data/themes/topics/topic-details/GHO/ncd-risk-factors. 

In [145]:
#Renaming the columns for hypertension 

hypertension.rename(columns={"ParentLocation":"Region","Location":"Country","Period":"Year", "Dim1":"Gender", 
                            "FactValueNumeric":"Hypertension_Value",
                            "FactValueNumericLow":"Hypertension_Low_Value",
                            "FactValueNumericHigh":"Hypertension_High_Value",
                             "Value":"Range"
                            }, inplace=True)
hypertension.head()


Indicator  Region       Country  Year  \
0  Mean total cholesterol,  age-standardized  Africa       Lesotho  2018   
1  Mean total cholesterol,  age-standardized  Africa        Rwanda  2018   
2  Mean total cholesterol,  age-standardized  Africa        Malawi  2018   
3  Mean total cholesterol,  age-standardized  Africa       Nigeria  2018   
4  Mean total cholesterol,  age-standardized  Africa  Burkina Faso  2018   

  Gender  Cholesterol_Value  Cholesterol_Low_Value  Cholesterol_High_Value  \
0   Male                3.6                    3.1                     4.1   
1   Male                3.6                    3.2                     4.1   
2   Male                3.6                    3.3                     3.9   
3   Male                3.7                    3.1                     4.3   
4   Male                3.7                    3.2                     4.1   

           Range  
0  3.6 [3.1-4.1]  
1  3.6 [3.2-4.1]  
2  3.6 [3.3-3.9]  
3  3.7 [3.1-4.3]  
4  3.7 [3.2-4.1]

In [163]:
#Renaming the columns for cholesterol 

cholesterol.rename(columns={"ParentLocation":"Region","Location":"Country","Period":"Year", "Dim1":"Gender", 
                            "FactValueNumeric":"Cholesterol_Value",
                            "FactValueNumericLow":"Cholesterol_Low_Value",
                            "FactValueNumericHigh":"Cholesterol_High_Value",
                            "Value":"Range"
                            }, inplace=True)
cholesterol.head()

Indicator  Region       Country  Year  \
0  Mean total cholesterol,  age-standardized  Africa       Lesotho  2018   
1  Mean total cholesterol,  age-standardized  Africa        Rwanda  2018   
2  Mean total cholesterol,  age-standardized  Africa        Malawi  2018   
3  Mean total cholesterol,  age-standardized  Africa       Nigeria  2018   
4  Mean total cholesterol,  age-standardized  Africa  Burkina Faso  2018   

  Gender  Cholesterol_Value  Cholesterol_Low_Value  Cholesterol_High_Value  \
0   Male                3.6                    3.1                     4.1   
1   Male                3.6                    3.2                     4.1   
2   Male                3.6                    3.3                     3.9   
3   Male                3.7                    3.1                     4.3   
4   Male                3.7                    3.2                     4.1   

           Range  
0  3.6 [3.1-4.1]  
1  3.6 [3.2-4.1]  
2  3.6 [3.3-3.9]  
3  3.7 [3.1-4.3]  
4  3.7 [3.2-4.1]

In [266]:
#Renaming the columns for mortality 

mortality.rename(columns={"ParentLocation":"Region","Location":"Country","Period":"Year", "Dim1":"Gender", 
                            "Dim2":"Disease",
                            "FactValueNumeric":"Mortality_Value",
                            "FactValueNumericLow":"Mortality_Low_Value",
                            "FactValueNumericHigh":"Mortality_High_Value", 
                             "Value":"Range"}
                            , inplace=True)
mortality.head()

Region      Country  Year      Gender               Disease  \
0  South-East Asia  Timor-Leste  2019        Male     Diabetes mellitus   
1  South-East Asia       Bhutan  2019      Female     Diabetes mellitus   
2           Europe   Montenegro  2019  Both sexes  Respiratory diseases   
3           Africa   Seychelles  2019        Male   Malignant neoplasms   
4           Africa      Lesotho  2019        Male     Diabetes mellitus   

   Mortality_Value  Mortality_Low_Value  Mortality_High_Value            Range  
0            99.53                49.54                 167.8     100 [50-168]  
1           100.20                56.06                 163.0     100 [56-163]  
2           100.00                63.92                 145.0     100 [64-145]  
3            99.63                66.17                 144.1     100 [66-144]  
4          1003.00               580.00                1612.0  1003 [580-1612]

In [159]:
#Taking the hypertension dataframe and writing it to SQL table called 'hypertension_table'. 

hypertension.to_sql("hypertension_table", engine, index=False,if_exists='replace') 
#index false - used to not include the index from the dataframe and add it as columns in the SQL table
#if_exists - if the table exists already, then replace it 
hypertension_table_df= pd.read_sql_table("hypertension_table", engine) #naming SQL table 
hypertension_table_df.head() #dataframe 

Indicator    Region      Country  \
0  Prevalence of hypertension among adults aged 3...    Europe  Switzerland   
1  Prevalence of hypertension among adults aged 3...  Americas         Peru   
2  Prevalence of hypertension among adults aged 3...  Americas       Canada   
3  Prevalence of hypertension among adults aged 3...  Americas         Peru   
4  Prevalence of hypertension among adults aged 3...    Europe        Spain   

   Year      Gender  Hypertension_Value  Hypertension_Low_Value  \
0  2019      Female                17.5                    11.9   
1  2019      Female                18.4                    15.3   
2  2019      Female                19.9                    16.0   
3  2019  Both sexes                20.7                    18.2   
4  2019      Female                20.8                    15.7   

   Hypertension_High_Value             Range  
0                     24.2  17.5 [11.9-24.2]  
1                     21.8  18.4 [15.3-21.8]  
2                     24.4  19.9 [16.0-24.4]  
3                     23.3  20.7 [18.2-23.3]  
4                     26.5  20.8 [15.7-26.5]

In [15]:
#Taking the cholesterol dataframe and writing it to SQL table called 'cholesterol_table'. 

cholesterol.to_sql("cholesterol_table", engine, index=False,if_exists='replace')
cholesterol_table_df = pd.read_sql_table("cholesterol_table", engine)
cholesterol_table_df.head()

Indicator  Region       Country  Year  \
0  Mean total cholesterol,  age-standardized  Africa       Lesotho  2018   
1  Mean total cholesterol,  age-standardized  Africa        Rwanda  2018   
2  Mean total cholesterol,  age-standardized  Africa        Malawi  2018   
3  Mean total cholesterol,  age-standardized  Africa       Nigeria  2018   
4  Mean total cholesterol,  age-standardized  Africa  Burkina Faso  2018   

  Gender  Cholesterol_Value  Cholesterol_Low_Value  Cholesterol_High_Value  \
0   Male                3.6                    3.1                     4.1   
1   Male                3.6                    3.2                     4.1   
2   Male                3.6                    3.3                     3.9   
3   Male                3.7                    3.1                     4.3   
4   Male                3.7                    3.2                     4.1   

           Range  
0  3.6 [3.1-4.1]  
1  3.6 [3.2-4.1]  
2  3.6 [3.3-3.9]  
3  3.7 [3.1-4.3]  
4  3.7 [3.2-4.1]

In [17]:
#Taking the mortality dataframe and writing it to SQL table called 'mortality_table'. 

mortality.to_sql("mortality_table", engine, index=False,if_exists='replace')
mortality_table_df= pd.read_sql_table("mortality_table", engine)
mortality_table_df.head()

Region      Country  Year      Gender               Disease  \
0  South-East Asia  Timor-Leste  2019        Male     Diabetes mellitus   
1  South-East Asia       Bhutan  2019      Female     Diabetes mellitus   
2           Europe   Montenegro  2019  Both sexes  Respiratory diseases   
3           Africa   Seychelles  2019        Male   Malignant neoplasms   
4           Africa      Lesotho  2019        Male     Diabetes mellitus   

   Mortality_Value  Mortality_Low_Value  Mortality_High_Value            Range  
0            99.53                49.54                 167.8     100 [50-168]  
1           100.20                56.06                 163.0     100 [56-163]  
2           100.00                63.92                 145.0     100 [64-145]  
3            99.63                66.17                 144.1     100 [66-144]  
4          1003.00               580.00                1612.0  1003 [580-1612]

In [19]:
pd.read_sql_query("SHOW TABLES;", engine)

#We can see that the three tables are now under engine 

Tables_in_milestone
0     cholesterol_table
1   cholesterol_table_1
2    hypertension_table
3  hypertension_table_1
4       mortality_table

### Cleaning Data and Preparing SQL Tables

Up until this point, we have only prepared our tables for our queries which are found below. Preparation included renaming the columns to appropriate names as they will be called in our queries. This included adding underscores in replaced of a space as this technique can avoid any miscallings. We have also converted the three dataframes into an SQL table and have assigned the following names: cholesterol_table, hypertertension_table, and mortality_table. More information about the topic is found in the query discussions below. Please note that the discussions regarding the queries are broken up into 5 parts (Part 1, Part 2, Part 3, Part 4 and Part 5). 

# Part 1

In [151]:
#Query: 1.A - Selecting hypertension data specific to Canada in the years 2000 to 2018. 

query_1 = """ SELECT * 
FROM hypertension_table 
WHERE Indicator = 'Prevalence of hypertension among adults aged 30-79 years, age-standardized'
  AND Year >= '2000' 
  AND Year <= '2018' 
  AND Country = 'Canada';"""
pd.read_sql_query(query_1, engine) 

#adding this table to a canadian specific dataset 

canada_hypertension=pd.read_sql_query(query_1, engine) 
canada_hypertension.to_sql("hypertension_table_1", engine, index=False, if_exists='replace')

canada_hypertension=pd.read_sql_table("hypertension_table_1", engine)
canada_hypertension.head()

     

Indicator    Region Country  Year  \
0  Prevalence of hypertension among adults aged 3...  Americas  Canada  2018   
1  Prevalence of hypertension among adults aged 3...  Americas  Canada  2018   
2  Prevalence of hypertension among adults aged 3...  Americas  Canada  2018   
3  Prevalence of hypertension among adults aged 3...  Americas  Canada  2017   
4  Prevalence of hypertension among adults aged 3...  Americas  Canada  2017   

       Gender  Hypertension_Value  Hypertension_Low_Value  \
0      Female                20.2                    16.8   
1  Both sexes                22.4                    19.7   
2        Male                24.5                    20.6   
3      Female                20.5                    17.4   
4  Both sexes                22.6                    20.2   

   Hypertension_High_Value             Range  
0                     24.1  20.2 [16.8-24.1]  
1                     25.2  22.4 [19.7-25.2]  
2                     28.6  24.5 [20.6-28.6]  
3                     24.0  20.5 [17.4-24.0]  
4                     25.2  22.6 [20.2-25.2]

In [29]:
#Query 1.B - Selecting cholesterol data specific to Canada in the years 2000 to 2018. 

query_1b = """ SELECT * 
FROM cholesterol_table 
WHERE Year >= '2000' 
  AND Year <= '2018' 
  AND Country = 'Canada';"""
pd.read_sql_query(query_1b, engine) 

canada_cholesterol=pd.read_sql_query(query_1b, engine) 
canada_cholesterol.to_sql("cholesterol_table_1", engine, index=False, if_exists='replace')

canada_cholesterol=pd.read_sql_table("cholesterol_table_1", engine)
canada_cholesterol.head()

Indicator    Region Country  Year  \
0  Mean total cholesterol,  age-standardized  Americas  Canada  2018   
1  Mean total cholesterol,  age-standardized  Americas  Canada  2018   
2  Mean total cholesterol,  age-standardized  Americas  Canada  2018   
3  Mean total cholesterol,  age-standardized  Americas  Canada  2017   
4  Mean total cholesterol,  age-standardized  Americas  Canada  2017   

       Gender  Cholesterol_Value  Cholesterol_Low_Value  \
0        Male                4.6                    4.3   
1  Both sexes                4.6                    4.4   
2      Female                4.7                    4.4   
3        Male                4.6                    4.3   
4  Both sexes                4.7                    4.5   

   Cholesterol_High_Value          Range  
0                     4.8  4.6 [4.3-4.8]  
1                     4.8  4.6 [4.4-4.8]  
2                     5.0  4.7 [4.4-5.0]  
3                     4.8  4.6 [4.3-4.8]  
4                     4.8  4.7 [4.5-4.8]

In [31]:
#Query 1.C - Selecting mortality data specific to Canada in the years 2000 to 2018. 

query_1c = """ SELECT *
FROM mortality_table 
WHERE Disease = 'Cardiovascular diseases'
 AND Year >= '2000' 
 AND Year <= '2018' 
 AND Country = 'Canada';"""
 
pd.read_sql_query(query_1c, engine) 

canada_mortality=pd.read_sql_query(query_1c, engine) 
canada_mortality.to_sql("mortality_table_1", engine, index=False, if_exists='replace')
canada_mortality=pd.read_sql_table("mortality_table_1", engine)
canada_mortality.head()

Region Country  Year      Gender                  Disease  \
0  Americas  Canada  2018      Female  Cardiovascular diseases   
1  Americas  Canada  2018        Male  Cardiovascular diseases   
2  Americas  Canada  2018  Both sexes  Cardiovascular diseases   
3  Americas  Canada  2017      Female  Cardiovascular diseases   
4  Americas  Canada  2017        Male  Cardiovascular diseases   

   Mortality_Value  Mortality_Low_Value  Mortality_High_Value  \
0          34074.0              25338.0               42010.0   
1          36074.0              29449.0               43330.0   
2          70147.0              54787.0               85340.0   
3          34501.0              25671.0               42446.0   
4          36871.0              30187.0               44097.0   

                 Range  
0  34074 [25338-42010]  
1  36074 [29449-43330]  
2  70147 [54787-85340]  
3  34501 [25671-42446]  
4  36871 [30187-44097]

### Part 1 Discussion: Query 1.A., Query 1.B., and Query 1.C. 

In the DATA 604 project, my group is exploring noncommunicable diseases within Canada and we are analyzing the impact of lifestyle behaviors on metabolic risk factors, specifically: obesity, hypertension, cholesterol, physical inactivity, and diabetes. My role is to explore one of our guiding questions "Cardiovascular Disease: How do cholesterol levels influence the prevalence of hypertension? How do these factors correlate to the numbers of deaths attributed to cardiovascular disease?". Cholesterol nowadays is easily accessible through processed and highly saturated fat foods. It is important to monitor cholesterol levels as an excess of this bulky substance can add stress to the heart causing the risk of heart issues or even heart failure (American Heart Association, 2024). The queries 1.A, 1.B, and 1.C, are all specific to retrieving data specific to Canada as our datasets are derived from the World Health Organization (W.H.O) therefore they list countries from all over the world.  

Query 1.A: The hypertension dataset initially had 34,980 rows as it was listing hypertension values from countries all over the world. In this query, the country 'Canada' was selected as our aim is to focus on metabolic risk factors in Canada. The years 2000 to 2018 were selected as our cholesterol dataset (which we are merging with later) only has records up to the 2018 year therefore we must on retrieve information up to the 2018 year for all datasets. Please note that hypertension is a percentage and it represents how much the population is experiencing hypertension in that year. These hypertension values are also age-standardized and are speciifc to the ages 30 to 79. 

Query 1.B: The cholesterol dataset initally had 22,347 rows due to the same reason for Query 1.A, it was obtaining global values therefore we selected only Canada. The years 2000 to 2018 were extracted. The W.H.O states that values above 5.0 mmol/L are considered 'high cholesterol levels'. 

Query 1.C: The mortality dataset is included as we would like to determine the trend in mortality when the population's cholesterol levels and hypertension is high. The mortality dataset initally had 43,920 rows therefore we created this query to select the rows specific to Canada. Furthermore, we are only interested in the link between cholesterol and hypertension, we only extract mortality values that are specific to cardiovascular diseases. Please note that the cardiovascular diseases listed goes beyond both the hypertension and cholesterol risk factors, meaning that the value includes other conditions related to cardiovascular health (i.e. cancer, stroke..etc.).

Overall, the purpose and significance of the queries in Part 1 was to select Canada from the global datasets and the years we would like to analyze (2000-2018). All queries were assigned to a table that specifically holds data specific to Canada (hypertension_table_1, cholesterol_table_1, and mortality_table_1). The purpose of this specific-Canada table is for further analysis as demonstrated in the queries listed for Part 2 and Part 3. 


# Part 2

In [177]:
#Query 2.A: Joining Cholesterol and Hypertension to make a single table. 

query_2a = """ 
SELECT 
    h.Region,
    h.Country, 
    h.Year, 
    h.Gender, 
    h.Hypertension_Value, 
    c.Cholesterol_Value 

FROM 
    hypertension_table_1 h
JOIN 
    cholesterol_table_1 c 
ON
    h.Year = c.Year AND h.Gender = c.Gender;
"""

combined_a = pd.read_sql_query(query_2a, engine)
combined_a.head()
      

Region Country  Year      Gender  Hypertension_Value  Cholesterol_Value
0  Americas  Canada  2018        Male                24.5                4.6
1  Americas  Canada  2018  Both sexes                22.4                4.6
2  Americas  Canada  2018      Female                20.2                4.7
3  Americas  Canada  2017        Male                24.8                4.6
4  Americas  Canada  2017  Both sexes                22.6                4.7

In [181]:
#Query 2.B - Joining the combined Hypertension and Cholesterol data with Mortality data to make a table. 

query_2b = """ 
SELECT 
    h.Region,
    h.Country, 
    h.Year, 
    h.Gender, 
    h.Hypertension_Value, 
    c.Cholesterol_Value,
    m.Mortality_Value,
    m.Disease

FROM 
    hypertension_table_1 h
JOIN 
    cholesterol_table_1 c 
ON
    h.Year = c.Year AND h.Gender = c.Gender
JOIN 
    mortality_table_1 m
ON 
    h.Year = m.Year AND h.Gender = m.Gender;
"""

combined_b = pd.read_sql_query(query_2b, engine)
combined_b.head()

canada_data=pd.read_sql_query(query_2b, engine) 
canada_data.to_sql("canada_data", engine, index=False, if_exists='replace')
canada_data=pd.read_sql_table("canada_data", engine)
canada_data.head()


Region Country  Year      Gender  Hypertension_Value  Cholesterol_Value  \
0  Americas  Canada  2018      Female                20.2                4.7   
1  Americas  Canada  2018        Male                24.5                4.6   
2  Americas  Canada  2018  Both sexes                22.4                4.6   
3  Americas  Canada  2017      Female                20.5                4.7   
4  Americas  Canada  2017        Male                24.8                4.6   

   Mortality_Value                  Disease  
0          34074.0  Cardiovascular diseases  
1          36074.0  Cardiovascular diseases  
2          70147.0  Cardiovascular diseases  
3          34501.0  Cardiovascular diseases  
4          36871.0  Cardiovascular diseases

### Part 2 Discussion  - Query 2.A and 2.B 

In this Part 2 of the queries, we are joining the Canada-specific tables that were made in Part 1. The purpose and significance of this step was to create a simple table that contains all the values we want for our analysis. Instead of having to look at three tables, we can focus all of our attention on this one canada_data table and we can now interpret a trend just by looking. 

Query 2.A: The first table created was to look at the relationship between cholesterol and hypertension. I made sure that the two tables were being joined and aligned based on the Year and the Gender as the Country is the same for all rows. 

Query 2.B: The purpose of the second table is to expand the first table by adding data from the Canada-specific mortality table. This table is saved to a new SQL table called 'canada_data' as it contains all the necessary information we need to start determing trends between cholesterol, hypertension, and mortality. This table was also aligned based on the Year and Gender which was done by these lines of code `h.Year = c.Year AND h.Gender = c.Gender` and  `h.Year = m.Year AND h.Gender = m.Gender;` where 'h' represents hypertension table, 'c' represents cholesterol tables, and 'm' represents mortality table. 


# Part 3

In [184]:
#Query 3: Observing high hypertension and cholesterol values in the years 2000 to 2018. 

query_3 = """ SELECT *
FROM canada_data
WHERE Gender != 'Both sexes'
  AND Cholesterol_Value > 5.0;"""
pd.read_sql_query(query_3, engine) 

#mortality is at its greatest for cardiovascular diseases 
#when male and female populations reached their highest values 

Region Country  Year  Gender  Hypertension_Value  Cholesterol_Value  \
0  Americas  Canada  2003  Female                24.6                5.1   
1  Americas  Canada  2003    Male                29.4                5.1   
2  Americas  Canada  2002    Male                29.8                5.1   
3  Americas  Canada  2002  Female                24.9                5.1   
4  Americas  Canada  2001    Male                30.3                5.1   
5  Americas  Canada  2001  Female                25.3                5.1   
6  Americas  Canada  2000    Male                30.8                5.2   
7  Americas  Canada  2000  Female                25.6                5.2   

   Mortality_Value                  Disease  
0          36693.0  Cardiovascular diseases  
1          36704.0  Cardiovascular diseases  
2          36302.0  Cardiovascular diseases  
3          37439.0  Cardiovascular diseases  
4          36613.0  Cardiovascular diseases  
5          37332.0  Cardiovascular diseases  
6          37520.0  Cardiovascular diseases  
7          37878.0  Cardiovascular diseases

### Part 3 Discussion - Query 3

The W.H.O has defined that 'high cholesterol levels' are values that exceed beyond 5.0 mmol/L. Therefore the purpose of this query is to answer the questions, which years did the Canada population experience high cholesterol? And what was the impact on the population's hypertension value and mortality? This query specifies that we are looking for cholesterol levels beyond 5.0 mmol/L for both the male and female population as demonstrated in this line `WHERE Gender != 'Both sexes' AND Cholesterol_Value > 5.0`. Overall, Part 3 query is strongly related to the project and guiding question.

From the output, the analysis shows that in the year 2000 had the highest record of cholesterol (between the years 2000 and 2018) for both the male and female population experienced as both obtained a level of 5.2 mmol/L. Interestingly enough, the year 2000 also experienced the highest mortality value (37,878 for females and 37,520 for males) compared to other years that have a cholesterol value beyond 5.0 mmol/L. In addition, 25.6% of the female population and 30.8% of the male population had hypertension. These are also the largest values for hypertension compared to the other years listed. 

It is important to note that correlation is not causation, and this query only examines the years from 2000 to 2003 therefore further analysis is required. This can included adding more Years and observing other metabolic risk factors related to cardiovascular health. Nonetheless, this output can suggest that cholesterol and hypertension may be linked when it comes to mortality. It is interesting to see that only the years 2000, 2001, 2002 and 2003 experience high cholesterol. Cholesterol values may have decreased since 2003 due to advanced research, more awareness, or accessibility to more healthcare facilities.  

# Part 4

In [235]:
#Query 4.A: Ranking countries based on hypertension (from highest to lowest).  

query_4a = """ SELECT Country, 
    AVG(Hypertension_Value) AS Average_Hypertension_Value,
    RANK() OVER (ORDER BY AVG(Hypertension_Value) DESC) AS 'Rank'
FROM 
    hypertension_table
WHERE 
    Gender = 'Both sexes' 
    AND Year BETWEEN 2000 AND 2018
GROUP BY 
    Country
ORDER BY 
    'Rank';"""              

pd.read_sql_query(query_4a, engine)

query_4a_rank = pd.read_sql_query(query_4a, engine)
query_4a_rank

#Rank is the new column that shows the ranking of the countries based on hypertension value 

Country  Average_Hypertension_Value  Rank
0               Paraguay                   54.350000     1
1              Lithuania                   53.473684     2
2                Belarus                   52.307895     3
3    Republic of Moldova                   51.102632     4
4                Hungary                   51.094737     5
..                   ...                         ...   ...
189              Eritrea                   24.239474   190
190      Solomon Islands                   23.589474   191
191             Cambodia                   22.686842   192
192     Papua New Guinea                   22.289474   193
193                 Peru                   19.257895   194

[194 rows x 3 columns]

In [217]:
#Determining the rank of Canada from 2000 to 2018 based on hypertension.

canada_rank = query_4a_rank[query_4a_rank['Country'] == 'Canada']

average_rank = canada_rank['Rank'].mean()
average_value = canada_rank['Average_Hypertension_Value'].mean()

rounded_average_rank = round(average_rank)
rounded_average_value = round(average_value)

print(f"The average rank for Canada based on hypertension from 2000 to 2018 is {rounded_average_rank} with an average value of {rounded_average_value}.")

The average rank for Canada based on hypertension from 2000 to 2018 is 186 with an average value of 27.


In [233]:
#Query 4.b: Ranking countries based on cholesterol (from highest percentage to lowest percentage).  

query_4b = """ SELECT Country, 
    AVG(Cholesterol_Value) AS Average_Cholesterol_Value,
    RANK() OVER (ORDER BY AVG(Cholesterol_Value) DESC) AS 'Rank'
FROM 
    cholesterol_table
WHERE 
    Gender = 'Both sexes' 
    AND Year BETWEEN 2000 AND 2018
GROUP BY 
    Country
ORDER BY 
    'Rank';"""               

pd.read_sql_query(query_4b, engine)

query_4b_rank = pd.read_sql_query(query_4b, engine)
query_4b_rank


Country  Average_Cholesterol_Value  Rank
0       Lithuania                   5.373684     1
1          Serbia                   5.300000     2
2          Latvia                   5.294737     3
3         Croatia                   5.257895     4
4           Malta                   5.252632     5
..            ...                        ...   ...
186       Lesotho                   3.800000   187
187       Nigeria                   3.757895   188
188  Burkina Faso                   3.757895   188
189        Rwanda                   3.710526   190
190        Malawi                   3.652632   191

[191 rows x 3 columns]

In [227]:
canada_rank_b = query_4b_rank[query_4b_rank['Country'] == 'Canada']

average_rank_b = canada_rank_b['Rank'].mean()
average_value_b = canada_rank_b['Average_Cholesterol_Value'].mean()

rounded_average_rank_b = round(average_rank_b)
rounded_average_value_b = round(average_value_b)

print(f"The average rank for Canada based on cholesterol from 2000 to 2018 is {rounded_average_rank_b} with an average value of {rounded_average_value_b}.")

The average rank for Canada based on cholesterol from 2000 to 2018 is 54 with an average value of 5.


### Part 4 Discussion - Query 4.A and 4.B 

The purpose of the queries listed in Part 4 is to compare Canada to the rest of the world. This is not specific to our guiding question however it is still important to determine where the country stands when it comes to health. The queries that were made rank the world countries based on their hypertension and cholesterol values in descending order (done by `DESC`). We are still looking at the years 2000 to 2018 as these are the specific years that are being investigating in Canada. This is done by inclduing the line `Year BETWEEN 2000 AND 2018`. The data derived from these query are for 'Both sexes' as stated in this line of code `Gender = 'Both sexes'`. This category simplifies our search as it takes the seperate hypertension values of both the female and male population into account. 

Query 4.A: This query is specific to the average hypertension value and it the output determinies which country's population experienced the highest hypertension percentage from combining each year from 2000 to 2018. This was done by creating a new column called Average_Hyptertenion_Value using the `AS` function . The line of code is `AVG(Hypertension_Value) AS Average_Hypertension_Value'`. Another column called 'Rank' was made and the table was orderd based on this column as stated by `ORDER BY 'Rank'`. Please note that this data is age standardized as we are investigating the years between 30 to 79 years old. Paraguay has the highest averaged hypertension value of 54.35% in the years from 2000 to 2018. The average rank for Canada based on hypertension from 2000 to 2018 is 186 out of 194 with an average value of 27% for both sexes. This positively shows that Canada is ranked rather low for hypertension compared to most of the rest of the world. 

Query 4.B: Query 4.B is specific to the average cholesterol levels by combing the cholesterol values over 18 years (2000-2018). This query is similar to 4.A as it also created a new column called Average_Cholesterol_Value using the `AS` function . The line of code is `AVG(Cholesterol_Value) AS Average_Cholesterol_Value'`. This data was also orderd by the 'Rank' column using `ORDER BY`. Please note that this data is age standardized as we are investigating the ages 18 and older. Lithuania came out on top for having the highest averaged cholesterol level over 18 years with a value of 5.37. The average rank for Canada for cholesterol level from 2000 to 2018 is 54 out of 191 with an average value of 5.0 mmol/L for both sexes. This is a moderate result for Canada as its ranked higher compared to our output for hypertension. 

# Part 5

In [250]:
#Query 5.A: Taking the average of female and male hypertension values by region. 

query_5a = """ SELECT Region, Gender, 
AVG(Hypertension_Value) AS Average_Hypertension_Value,
RANK() OVER (ORDER BY AVG(Hypertension_Value) ASC) AS 'Rank' 
FROM hypertension_table
WHERE Gender = 'Male' OR Gender = 'Female'
AND Year BETWEEN 2000 AND 2018
GROUP BY Region, Gender
ORDER BY 'Rank'; """

hypertension_avg_data = pd.read_sql_query(query_5a, engine)
hypertension_avg_data.head(15)

Region  Gender  Average_Hypertension_Value  Rank
0         South-East Asia    Male                   30.751818     1
1         Western Pacific  Female                   32.085837     2
2         South-East Asia  Female                   32.307656     3
3         Western Pacific    Male                   33.836121     4
4                  Africa    Male                   35.186028     5
5   Eastern Mediterranean  Female                   36.025837     6
6   Eastern Mediterranean    Male                   36.091515     7
7                Americas    Male                   36.911157     8
8                Americas  Female                   37.240936     9
9                  Africa  Female                   37.598208    10
10                 Europe  Female                   38.951393    11
11                 Europe    Male                   44.610458    12

In [252]:
#Query 5.b: Taking the average of female and male cholesterol values by region.

query_5b = """ SELECT Region, Gender, 
AVG(Cholesterol_Value) AS Average_Cholesterol_Value,
RANK() OVER (ORDER BY AVG(Cholesterol_Value) ASC) AS 'Rank'
FROM cholesterol_table 
WHERE Gender = 'Male' OR Gender = 'Female'
AND Year BETWEEN 2000 AND 2018
GROUP BY Region, Gender 
ORDER BY 'Rank'; """ 

cholesterol_avg_data = pd.read_sql_query(query_5b, engine)
cholesterol_avg_data.head(15)

Region  Gender  Average_Cholesterol_Value  Rank
0                  Africa    Male                   3.920346     1
1                  Africa  Female                   4.170595     2
2         South-East Asia    Male                   4.313054     3
3         South-East Asia  Female                   4.531579     4
4   Eastern Mediterranean    Male                   4.535043     5
5                Americas    Male                   4.606374     6
6   Eastern Mediterranean  Female                   4.652130     7
7         Western Pacific    Male                   4.652327     8
8         Western Pacific  Female                   4.706238     9
9                Americas  Female                   4.724962    10
10                 Europe  Female                   5.005882    11
11                 Europe    Male                   5.142081    12

In [276]:
#Query 5.c: Determining the trend of mortality 

query_5c = """ SELECT Region, Gender, 
AVG(Mortality_Value) AS Average_Mortality_Value,
RANK() OVER (ORDER BY AVG(Mortality_Value) ASC) AS 'Rank'
FROM mortality_table 
WHERE Gender = 'Male' OR Gender = 'Female'
AND Disease = 'Cardiovascular diseases'
AND Year BETWEEN 2000 AND 2018
GROUP BY Region, Gender 
ORDER BY 'Rank'; """ 

mortality_trend_data = pd.read_sql_query(query_5c, engine)
mortality_trend_data.head(15)

Region  Gender  Average_Mortality_Value  Rank
0                  Africa    Male              4447.651590     1
1                  Africa  Female             10077.560806     2
2   Eastern Mediterranean    Male             11947.563470     3
3                Americas    Male             14284.597750     4
4                  Europe    Male             16958.288438     5
5   Eastern Mediterranean  Female             25918.040351     6
6                Americas  Female             27127.025152     7
7                  Europe  Female             46872.743789     8
8         Western Pacific    Male             62196.826702     9
9         South-East Asia    Male             74048.402045    10
10        Western Pacific  Female             98518.372030    11
11        South-East Asia  Female            127499.166029    12

### Part 5 Discussion - Query 5.A, 5.B and 5.C

The purpose of our Part 5 queries it to look at our values on a larger scale by comparing the bigger regions. This Part 5 is also not specific to our guiding question however it is interesting to look at which parts of the world are doing well and which parts need more help. This information is critical for an organization like the W.H.O. These queries output both the male and female population seperately with the code `WHERE Gender = 'Male' OR Gender = 'Female'`. The queries also grouped the information based on Region and Gender with the line `GROUP BY Region, Gender`. By analyzing these values we can do further research on which cultures have the best diet, physical activity, environment ...etc. By figuring out which cultures are thriving, health authorities can investigate and implement ideas that are inspired by healthier regions and improve their own country's health policy. Health authorities can also determine which public health factors to avoid by observing the regions that are not thriving. 

Query 5.A: This query called the averaged hypertension value over 18 years (2000-2018). We organized our table in an ascending order (`ASC`) to determine which region is supposedly 'the healthiest'. The region that experienced the lowest hypertension rate is South-East Asia for the male population as it obtained a value of 30.75. The healthiest female population is Western-Pacific region with an average of 32.09. 

Query 5.B: This query called the averaged cholesterol value over 18 years (2000-2018). We organized our table in an ascending order to determine which region is supposedly also 'the healthiest'. The region that experienced the lowest cholesterol level is in Africa for the male population as it obtained a value of 3.92. The healthiest female population is also in Africa with an average cholesterol level of 4.17 in the years between 2000-2018.

Query 5.C: This query was coded to investigate the average mortality value over 18 years (2000-2018). The region with the lowest mortality rate is Africa with a value of 4,447 for the male population. This region of Africa also derived the lowest mortality value for the female population as it obtained a value of 10,167 which is significantly higher than the male population. This is an interesting trend between the male and female population in Africa as the values are very different, this trend would require further research to determine the cause behind this drastic difference since the hypertension and cholesterol levels for the African male and female population are not that different from one another. 

Overall, compared to the rest of the world, the America region (which includes Canada) did experience high hypertension for both the male and female population. This would be an interesting topic to analyze which factors contribute to higher hypertension in the Americas compared to the rest of the world. Cholesterol level in the Americas was high for the female population (with only Europe edging ahead by 0.3 with an average value of 5.0 versus 4.7). The male population was ranked at a better position at 5th versus 9th for females. Finally, the mortality value for the Americas was ranked at 4th (males) and 8th (females). We can further our analysis for the tables by investigating more factors such as environment, employment rate, income...etc to see what lifestyle behaviour contribute to higher metabolic risk.